plugins/inputs/ssl/check_ssl.go

```diff
@@ -0,0 +1,102 @@
+package ssl
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"net"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type CheckExpire struct {
+	// Server to check
+	Servers []string
+
+	// Timeout in seconds. 0 means no timeout
+	Timeout string
+}
+
+// Description returns the plugin Description
+func (c *CheckExpire) Description() string {
+	return "time left until SSL cert is expired"
+}
+
+var sampleConfig = `
+  ## server name list default [] )
+  servers = ["github.com:443"]
+  ## Set timeout (default 5 seconds)
+  timeout = "5s"
+`
+
+// SampleConfig returns the plugin SampleConfig
+func (c *CheckExpire) SampleConfig() string {
+	return sampleConfig
+}
+
+// Connect to server and retrieve chain certificates
+func (c *CheckExpire) checkHost(server string) ([]*x509.Certificate, error) {
+
+	tout, _ := time.ParseDuration(c.Timeout)
```

Comment:
Probably shouldn't discard the error. If the user puts `5 tocks`, we should throw a "can't parse" error instead of silently assuming `tout = 0`.
It might also be good to do the parsing just once (in `Gather()`). The value won't change between servers.